db/db_with_timestamp_basic_test.cc

```diff
@@ -774,6 +774,36 @@ TEST_F(DBBasicTestWithTimestamp, BatchWriteAndMultiGet) {
   Close();
 }
 
+TEST_F(DBBasicTestWithTimestamp, MultiGetNoReturnTs) {
+  Options options = CurrentOptions();
+  options.env = env_;
+  const size_t kTimestampSize = Timestamp(0, 0).size();
+  TestComparator test_cmp(kTimestampSize);
+  options.comparator = &test_cmp;
+  DestroyAndReopen(options);
+  WriteOptions write_opts;
+  std::string ts_str = Timestamp(1, 0);
+  Slice ts = ts_str;
+  write_opts.timestamp = &ts;
+  ASSERT_OK(db_->Put(write_opts, "foo", "value"));
+  ASSERT_OK(db_->Put(write_opts, "bar", "value"));
```

Comment:
The failure is more interesting if the key is big enough to be mistaken for holding a timestamp: wrong result rather than assertion failure. I threw 16 X's after foo and bar.

Even better: I reproduced a crash:

    ASSERT_OK(db_->Put(write_opts, "foo", "value"));
    ASSERT_OK(db_->Put(write_opts, "bar", "value"));
    // Don't mistake trailing 16 chars in user key as timestamp
    ASSERT_OK(db_->Put(write_opts, "fooXXXXXXXXXXXXXXXX", "value"));
    ASSERT_OK(db_->Put(write_opts, "barXXXXXXXXXXXXXXXX", "value"));
    ...
    Slice keys[] = {"fooXXXXXXXXXXXXXXXX", "barXXXXXXXXXXXXXXXX"};

Reply:
I updated the test adding the two longer keys as shown above, but could not see a crash. Could you share the test that caused the crash?

Reply:
https://github.com/pdillinger/rocksdb/tree/fix-nullptr-multiget

Reply:
Can you please include a comment for why the 16 x's, as I did?

Reply:
Okay, I thought you reproduced a crash with this PR's fix. Will update.